packages/main/src/Table.js

```diff
@@ -446,6 +491,82 @@ class Table extends UI5Element {
 		this.fireEvent("load-more");
 	}
 
+	_handleSingleSelect(event) {
+		const row = this.getRowParent(event.target);
+		if (!row.selected) {
+			const previouslySelectedRows = this.rows.filter(item => item.selected);
+			this.rows.forEach(item => {
+				if (item.selected) {
+					item.selected = false;
+				}
+			});
+			row.selected = true;
+			this.fireEvent("selection-change", {
+				selectedRows: [row],
+				previouslySelectedRows,
+			});
+		}
+	}
+
+	_handleMultiSelect(event) {
+		const row = this.getRowParent(event.target);
+		const previouslySelectedRows = this.rows.filter(item => item.selected);
+
+		row.selected = !row.selected;
+
+		const selectedRows = this.rows.filter(item => item.selected);
+
+		if (selectedRows.length === this.rows.length) {
+			this._allRowsSelected = true;
+		} else {
+			this._allRowsSelected = false;
+		}
+
+		this.fireEvent("selection-change", {
+			selectedRows,
+			previouslySelectedRows,
+		});
+	}
+
+	_handleSelect(event) {
+		const mappings = {
```

Comment:
This object will be re-created each time we enter the function

you can do this
```js
this[`_handle${this.mode}`]](event);
```

or extract the "mappings" as static getter:
```js
static get SELECTION_HANDLERS_MAPPING() {
   return {
            "SingleSelect": "_handleSingleSelect",
			"MultiSelect": "_handleMultiSelect",
    }
}
```